### PR TITLE
Skip integration test for draft PRs

### PR DESCRIPTION
**Pull request by JostMigenda**
In #128, we changed the GitHub workflow so that integration test would not run on draft PRs. Unfortunately, with the solution we found back then, it required manual activity by a developer to run; so it did not run reliably on PRs, which has caused some problems recently.

After looking at the GitHub documentation some more, I think I’ve found a better way to achieve the same thing. I will test in this PR.

### CODE REVIEW

**Comment by JostMigenda** [comment]
In [728da55](https://github.com/SNEWS2/snewpy/pull/227/commits/728da555e9ff512dfac3ef796f66c7f893dd9833) you can see nicely that the “Integration Tests / build” job is skipped for this draft PR.

**Comment by JostMigenda** [comment]
Only downside now is that the integration test doesn’t re-run if a draft is marked as ready for review. But that should be easy to fix in the following commit

(FWIW, GitHub documentation for the `pull_request` event is [here (for activity types)](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) and [here (for the format of `github.event.pull_request`)](https://docs.github.com/en/rest/pulls/pulls#get-a-pull-request).)

**Comment by JostMigenda** [comment]
Before we review and merge this, one more question: As part of the integration test, we currently run a timing benchmark and publish the results in a comment on the PR. (@Sheshuk originally added that in https://github.com/SNEWS2/snewpy/pull/132/commits/3bb7899c7a01b524d0b1fe499569b8c16d5820c8)

Over the past year, I don’t remember us ever using the results of that benchmark, because they are so unreliable—it’s random chance whether the workflow runs on a GitHub worker machine that happens to be fast or slow. (E.g. in the comment above, if you view the edit history you can see that the mean time was between 3.49 and 5.47 for the exact same commit.) Instead, we run manual benchmarks anyway whenever we discuss performance improvements.
I would therefore suggest we remove this benchmark (and the workflow that publishes the results).

**Review by mcolomermolla — COMMENTED** [review]
I agree with removing benchmark timing from integration test, it is indeed completely randomly depending on the machine. Then, it can be merged.

**Comment by JostMigenda** [comment]
Okay, I’ve removed all the benchmarking code. @mcolomermolla, do you want to take a quick look and post an approving review if you’re happy with this?

**Review by mcolomermolla — APPROVED** [review]

**Pull request merged**
